File: Heat/Claims/HeatClaimsPrincipal.vb
Imports System.Security.Claims
Imports System.Security.Principal

Public Class HeatClaimsPrincipal
    Inherits ClaimsPrincipal

    Sub New(username As String)
        Dim gi As GenericIdentity = New GenericIdentity(username, "Heat Custom authentication")
        Dim ci As ClaimsIdentity = New ClaimsIdentity(gi)

        Me.AddIdentity(ci)
    End Sub


End Class

File: Heat.ConvertedToC#/Claims/HeatClaimsPrincipal.cs
using System.Security.Claims;
using System.Security.Principal;
namespace Heat
{

    public class HeatClaimsPrincipal : ClaimsPrincipal
	{

		public HeatClaimsPrincipal(string userName)
		{
			GenericIdentity gi = new GenericIdentity(userName, "Heat Custom authentication");
			ClaimsIdentity ci = new ClaimsIdentity(gi);

			this.AddIdentity(ci);
		}


	}
}
